library/std/src/os/uefi/ffi.rs

```diff
@@ -0,0 +1,73 @@
+use crate::ffi::{OsStr, OsString};
+use crate::sealed::Sealed;
+use crate::sys_common::ucs2;
+
+#[unstable(feature = "uefi_std", issue = "100499")]
+pub trait OsStrExt: Sealed {
+    /// This function does not do any allocation
+    fn to_ucs2<'a>(&'a self) -> ucs2::EncodeUcs2<'a>;
+
+    /// Creates a UCS-2 Vec which can be passed for FFI.
+    /// Note: This function will replace `NULL` and other characters which are not valid in UEFI
+    /// strings with `std::sys_common::ucs::Ucs2Char::REPLACEMENT_CHARACTER`
```

Comment:
I don't think this method should be part of the public API so it's perhaps a moot point, but `std::sys_common` is fully internal so we should not talk about it in documentation -- `char::REPLACEMENT_CHARACTER as u16` is probably clear enough.